extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java

```diff
@@ -575,19 +576,50 @@ DataFormat dataFormat() {
              */
             static final class Json extends KeyOrValueSpec {
                 final ColumnDefinition<?>[] columnDefinitions;
-                final Map<String, String> fieldNameToColumnName;
+                final Map<String, String> fieldToColumnName;
 
                 private Json(
                         final ColumnDefinition<?>[] columnDefinitions,
                         final Map<String, String> fieldNameToColumnName) {
                     this.columnDefinitions = columnDefinitions;
-                    this.fieldNameToColumnName = fieldNameToColumnName;
+                    this.fieldToColumnName = mapNonPointers(fieldNameToColumnName);
                 }
 
                 @Override
                 DataFormat dataFormat() {
                     return DataFormat.JSON;
                 }
+
+                private static Map<String, String> mapNonPointers(final Map<String, String> fieldNameToColumnName) {
+                    if (fieldNameToColumnName == null) {
+                        return null;
+                    }
+                    boolean needsMapping = false;
+                    for (Map.Entry<String, String> entry : fieldNameToColumnName.entrySet()) {
+                        final String key = entry.getKey();
+                        if (!key.startsWith("/")) {
+                            needsMapping = true;
+                            break;
+                        }
+                    }
+                    if (!needsMapping) {
+                        return fieldNameToColumnName;
+                    }
+                    final Map<String, String> ans = new HashMap<>(fieldNameToColumnName.size());
+                    for (Map.Entry<String, String> entry : fieldNameToColumnName.entrySet()) {
+                        final String key = entry.getKey();
+                        if (key.startsWith("/")) {
+                            ans.put(key, entry.getValue());
+                        } else {
+                            ans.put(mapTopLevelFieldNameToJsonPointerStr(key), entry.getValue());
+                        }
+                    }
+                    return ans;
+                }
+
+                public static String mapTopLevelFieldNameToJsonPointerStr(final String key) {
```

Comment:
Because the characters '~' (%x7E) and '/' (%x2F) have special
   meanings in JSON Pointer, '~' needs to be encoded as '~0' and '/'
   needs to be encoded as '~1' when these characters appear in a
   reference token.

Reply:
Documented.